File: solutions/python/anagram/1/anagram.py
def fix_word(word):
   l = list(word.lower())
   l.sort()
   return l 

def find_anagrams(word, candidates):
   out = []
   for candidate in candidates:
      if (candidate.lower() != word.lower() and fix_word(candidate) == fix_word(word)):
         out.append(candidate)
   return out

 

File: solutions/clojure/anagram/3/src/anagram.clj
(ns anagram
   (:require [clojure.string :as str]))

(defn fixit [s] 
  (str/join (sort (str/lower-case s))))

(defn anagrams-for [word prospect-list] 
   (->> prospect-list
        (filter #(and (not= (.toLowerCase word)(.toLowerCase %)) 
                      (= (fixit word) (fixit %)))))) 

